src/color-theme/script.js

```diff
@@ -1,58 +1,46 @@
+const storageKey = "data-scheme";
 const pressedButton = '[data-scheme][aria-pressed="true"]';
-const storageKey = 'theme-preference';
 
 const getColorPreference = () => {
-  if (localStorage.getItem(storageKey)) return localStorage.getItem(storageKey);
+  const storedScheme = localStorage.getItem(storageKey);
+  if (storedScheme) return storedScheme;
   else
-    return window.matchMedia('(prefers-color-scheme: dark)').matches
-      ? 'dark'
-      : 'light';
-};
-
-const setInitialScheme = () => {
-  const savedScheme = localStorage.getItem('data-scheme');
-  if (savedScheme) {
-    setScheme(savedScheme);
-  } else {
-    getColorPreference();
-    setScheme(theme.value);
-  }
+    return window.matchMedia("(prefers-color-scheme: dark)").matches
+      ? "dark"
+      : "light";
 };
 
 const setScheme = (scheme) => {
+  // Switch to the new scheme
+  document.documentElement.setAttribute("data-scheme", scheme);
```

Comment:
@jamesnw I think setting this attribute has to happen a bit earlier to prevent the flash that could happen if the `storageKey` is different from the OS setting. What do you think is the best way to adjust that?

Reply:
@jamesnw Thanks! This fixed the flash but the `aria-pressed` state wasn't updating since they aren't visible in the DOM until later (if you refreshed the page "Auto" was always pressed). 

I think we have to also call `setScheme ` again on load? I made that change  in https://github.com/oddbird/baseline-demos/pull/5/commits/61f95079345f334c89af2fc7593e99d6fe406257 .

Reply:
No worries! Changes look good 🙏🏽